# Copyright (C) 2005  Miriam Ruiz <little_miry@yahoo.es>
# Copyright (C) 2005  Dafydd Harries <daf@debian.org>
# Distributed under the same license as the game. See debian/copyright.

Index: rrootage-0.23a/src/attractmanager.c
===================================================================
--- rrootage-0.23a.orig/src/attractmanager.c	2007-08-29 19:48:31.000000000 +0000
+++ rrootage-0.23a/src/attractmanager.c	2007-08-29 19:50:13.000000000 +0000
@@ -64,9 +64,14 @@
   int i, j;
   int version;
   char *tmpname;
-  char name[128];
+  char name[256];
 
   tmpname = getenv("HOME");
+  if (strlen(tmpname) + strlen(PREF_FILE) >= 255) {
+    fprintf(stderr,"Preferences filename is too long.\n");
+    initHiScore();
+    return;
+  }
   strcpy(name, tmpname);
   strcat(name, PREF_FILE);
 
@@ -95,9 +100,13 @@
   FILE *fp;
   int i, j;
   char *tmpname;
-  char name[128];
+  char name[256];
 
   tmpname = getenv("HOME");
+  if (strlen(tmpname) + strlen(PREF_FILE) >= 255) {
+    fprintf(stderr,"Preferences filename is too long.\n");
+    return;
+  }
   strcpy(name, tmpname);
   strcat(name, PREF_FILE);
 
